include/deal.II/dofs/dof_tools.h

```diff
@@ -1437,6 +1437,39 @@ namespace DoFTools
                                          const std::set<types::boundary_id> &boundary_ids = std::set<types::boundary_id>());
 
   /**
+   * Extract all degrees of freedom (DoFs) that have support only within cells,
+   * for which @p predicate is <code>true</code>. The result is returned as
+   * an IndexSet.
+   *
+   * Consider the following FE space where predicate returns <code>true</code>
+   * for all cells on the left side:
+   *
+   * @image html extract_dofs_with_support_on.png
+   *
+   * This functions will return the union of all DoFs on those cells minus
+   * DoF 11, 13, 2 and 0; the result will be <code>[9,10], 12, [14,38]</code>.
+   * On the image above the returned DoFs are separated from the rest by the
+   * red line
+   *
+   * Essentially, the question this functions answers is the following:
+   * Given a subdomain with associated DoFs, what is the largest subset of
+   * these DoFs that are allowed to be non-zero such that after calling
+   * ConstraintMatrix::distribute() the resulting solution vector will have
+   * support only within the given domain. Here @p cm is the ConstraintMatrix
+   * containing hanging nodes constraints.
```

Comment:
Oh, I think I finally understand. What the function does is ask the following question: which shape function's support is *entirely contained* within the cells for which the predicate returns true!

So maybe a better name of the function is `extract_dofs_with_support_contained_within()`?